az-core/src/main/java/azkaban/utils/PropsUtils.java

```diff
@@ -93,6 +98,45 @@ public static Props loadProps(final Props parent, final File... propFiles) {
     }
   }
 
+  /**
+   * Load plugin properties
+   * @param pluginDir  plugin's Base Directory
+   * @param pluginType plugin type (viewer, trigger, alter)
+   * @return The properties
+   */
+  public static Props loadPluginProps(final File pluginDir, String pluginType) {
+    if (!pluginDir.exists()) {
```

Comment:
didn't get that. In general, I don't want to change any existing logic in this RB. It may need to throw exception. But, that's not the current implemented logic actually. I prefer not to changing any existing logic in this RB if I are not crystal clear the side-effect. Of course, we can modify the logic later if we believe it is necessary in future. What do you think?